dbt/include/trino/macros/adapters.sql

```diff
@@ -4,33 +4,25 @@
 -- - get_columns_in_relation
 
 {% macro trino__get_columns_in_relation(relation) -%}
-  {% call statement('get_columns_in_relation', fetch_result=True) %}
-      select
-          column_name,
-          case when regexp_like(data_type, 'varchar\(\d+\)') then 'varchar'
-               else data_type
-          end as data_type,
-          case when regexp_like(data_type, 'varchar\(\d+\)') then
-                  from_base(regexp_extract(data_type, 'varchar\((\d+)\)', 1), 10)
-               else NULL
-          end as character_maximum_length,
-          NULL as numeric_precision,
-          NULL as numeric_scale
-
-      from
-      {{ relation.information_schema('columns') }}
-
-      where
-        table_name = '{{ relation.name }}'
-        {% if relation.schema %}
-        and table_schema = '{{ relation.schema | lower }}'
-        {% endif %}
-      order by ordinal_position
-
-  {% endcall %}
+  {%- set sql -%}
+    describe {{ relation }}
+  {%- endset -%}
+  {%- set result = run_query(sql) -%}
+
+  {% set maximum = 10000 %}
+  {% if (result | length) >= maximum %}
+    {% set msg %}
+      Too many columns in relation {{ relation }}! dbt can only get
+      information about relations with fewer than {{ maximum }} columns.
+    {% endset %}
+    {% do exceptions.raise_compiler_error(msg) %}
+  {% endif %}
 
-  {% set table = load_result('get_columns_in_relation').table %}
-  {{ return(sql_convert_columns_in_relation(table)) }}
+  {% set columns = [] %}
```

Comment:
let's create a test containing all supported column types and see how this change performs.

Reply:
Not sure if that's really necessary as we are not testing the `describe` statement itself but only the output. The output of the macro contains `Type` and `Name` from the result set returned by `describe`. Are you worried about performance of `describe` vs `information_schema` lookups?

This method is already widely used within the integration tests (see basic adapter tests, incremental tests, ...).